client/Windows/wf_floatbar.c

```diff
@@ -170,51 +181,114 @@ static Button* floatbar_create_lock_button(FloatBar* floatbar,
+        {
+            if (floatbar->buttons[i] != NULL && x > floatbar->buttons[i]->x && x < floatbar->buttons[i]->x + floatbar->buttons[i]->w)
+            {
 				return floatbar->buttons[i];
+            }
+        }
+    }
 
 	return NULL;
 }
 
-static int floatbar_paint(FloatBar* floatbar, HDC hdc)
+static int floatbar_paint( FloatBar * const floatbar, const HDC hdc)
 {
 	int i;
 
 	/* paint background */
-	SelectObject(floatbar->hdcmem, floatbar->background);
-	StretchBlt(hdc, 0, 0, BACKGROUND_W, BACKGROUND_H, floatbar->hdcmem, 0, 0, BACKGROUND_W, BACKGROUND_H, SRCCOPY);
+  
+    GRADIENT_RECT gradientRect = { 0, 1 };
+    COLORREF rgbTop = RGB( 117, 154, 198 );
+    COLORREF rgbBottom = RGB( 6, 55, 120 );
+
+    const int top = 0;
+    int left = 0;
+    int bottom = BACKGROUND_H - 1;
+    int right = BACKGROUND_W - 1;
+    const int angleOffset = BACKGROUND_H - 1;
+
+    TRIVERTEX triVertext[2] = {
+        left,
+        top,
+        GetRValue( rgbTop ) << 8,
+        GetGValue( rgbTop ) << 8,
+        GetBValue( rgbTop ) << 8,
+        0x0000,
+        right,
+        bottom,
+        GetRValue( rgbBottom ) << 8,
+        GetGValue( rgbBottom ) << 8,
+        GetBValue( rgbBottom ) << 8,
+        0x0000
+    };
+
+    GradientFill( hdc, triVertext, 2, &gradientRect, 1, GRADIENT_FILL_RECT_V );
+
+    /* paint shadow */
+
+    HPEN hpen = CreatePen( PS_SOLID, 1, RGB( 71, 71, 71 ) );
+    HGDIOBJECT orig = SelectObject( hdc, hpen );
+
+    MoveToEx( hdc, left, top, NULL );
+    LineTo( hdc, left + angleOffset, bottom );
+    LineTo( hdc, right - angleOffset, bottom );
+    LineTo( hdc, right + 1, top - 1 );
+    DeleteObject( hpen );
+
+    hpen = CreatePen( PS_SOLID, 1, RGB( 107, 141, 184 ) );
+    SelectObject( hdc, hpen );
+
+    left += 1;
+    bottom -= 1;
+    right -= 1;
+
+    MoveToEx( hdc, left, top, NULL );
+    LineTo( hdc, left + ( angleOffset - 1 ), bottom );
+    LineTo( hdc, right - ( angleOffset - 1 ), bottom );
+    LineTo( hdc, right + 1, top - 1 );
+    DeleteObject( hpen );
+
+    SelectObject( hdc, orig );
+
+    DrawText( hdc, floatbar->wfc->window_title, wcslen( floatbar->wfc->window_title ), &floatbar->textRect, DT_CENTER | DT_VCENTER | DT_END_ELLIPSIS | DT_NOPREFIX | DT_SINGLELINE );
 
 	/* paint buttons */
+
 	for (i = 0; i < BTN_MAX; i++)
+    {
 		button_paint(floatbar->buttons[i], hdc);
+    }
 
 	return 0;
 }
 
-static int floatbar_animation(FloatBar* floatbar, BOOL show)
+static int floatbar_animation( FloatBar* const floatbar, const BOOL show)
 {
 	SetTimer(floatbar->hwnd, show ? TIMER_ANIMAT_SHOW : TIMER_ANIMAT_HIDE, 10, NULL);
 	floatbar->shown = show;
 	return 0;
 }
 
-LRESULT CALLBACK floatbar_proc(HWND hWnd, UINT Msg, WPARAM wParam, LPARAM lParam)
+LRESULT CALLBACK floatbar_proc(const HWND hWnd, const UINT Msg, const WPARAM wParam, const LPARAM lParam)
 {
 	static int dragging = FALSE;
 	static int lbtn_dwn = FALSE;
 	static int btn_dwn_x = 0;
+    static wfContext * wfc;
```

Comment:
Sorry.  Looks fine in Visual Studio and I kept the formatting consistent with existing code, but you probably are using different tab settings than we do so I didn't see any issues.  Again, just thought this might be useful for others.  Up against a deadline at work so I don't have time at the moment to reconfigure my editor and fix style issues.